### DISCUSSION TITLE
In Next.js setting a cookie in a server action causes wagmi to throw an error, and wagmi stays in "reconnecting"

### DISCUSSION THREAD

**Discussion by smauret** [wevm/wagmi#3607] [Q&A]
GM,

I have a Next.js application, and a client component calling a server action that sets a cookie and on next render I get this error and wagmi stays on status "reconnecting" (refreshing the page clears the issue):
 
![image](https://github.com/wevm/wagmi/assets/11448327/39a9d9ee-d649-4616-a539-ef25e7a50730)

I use `cookieToInitialState` with the same setup as in documentation.

My client component (asks for signature through wagmi and call the server action)  looks like this:

```js
"use client";
import { signMessage } from "@wagmi/core";
import { config } from "@/config";
import { postToken } from "@/app/lib/actions";
import { Button } from "@/app/ui/common/button";
import { useCallback } from "react";

export default function StartButton() {
  const signAuthMessage = useCallback(async () => {
    const signedMessage = await signMessage(config, { message: "msg" });

    const res = await postToken(signedMessage);
  }, []);

  return <Button onClick={signAuthMessage}>Start</Button>;
}

```

And the server action (uses previously signed message to call an api that will return a jwt token) looks like this:
```js
"use server";
export async function postToken(body: PostTokenBody) {
  try {
    const res = await fetch(`url`, {
      method: "POST",
      body: JSON.stringify(body),
    });
    const data = await res.json();
    cookies().set("jwtToken", data?.result?.token, { httpOnly: true });
    return data;
  } catch (err) {
    console.log("[POST tokens] error: ", err);
  }
}
```

A new cookie has been added after that, but it's unrelated to wagmi's cookie store, why does wagmi provider re-render and `setState` during hydration ?
Am I doing smth wrong ?

Thanks for the help,

**Comment by ducneuer232479** [comment]
Did you fix this issue ?

**Reply by smauret** [reply]
I opened an [issue](https://github.com/wevm/wagmi/issues/3629) for it and latest version of wagmi should have fixed it
